helix-term/src/commands.rs

```diff
@@ -5524,6 +5526,40 @@ fn shrink_selection(cx: &mut Context) {
     cx.editor.last_motion = Some(Motion(Box::new(motion)));
 }
 
+fn select_next_sibling(cx: &mut Context) {
+    let motion = |editor: &mut Editor| {
+        let (view, doc) = current!(editor);
+
+        if let Some(syntax) = doc.syntax() {
+            let text = doc.text().slice(..);
+
+            let current_selection = doc.selection(view.id);
+
+            let selection = object::select_next_sibling(syntax, text, current_selection);
+            doc.set_selection(view.id, selection);
+        }
+    };
+    motion(cx.editor);
+    cx.editor.last_motion = Some(Motion(Box::new(motion)));
+}
+
+fn select_prev_sibling(cx: &mut Context) {
+    let motion = |editor: &mut Editor| {
```

Comment:
I tried to factor these too so it's not duplicated everywhere except the sibling call, but I couldn't figure out how to make the borrow checker happy. I tried making a `fn(&Syntax, RopeSlice, &Selection)` argument, but it didn't like that the function argument gets captured by the motion closure.

Reply:
I'm a little afraid of the borrow checker :sweat_smile:, I'll give this a shot but I'm not optimistic

Reply:
Could do something like this: https://github.com/helix-editor/helix/blob/5549acdc7220f2d9341e6f0f958b9ea6226d7982/helix-term/src/commands.rs#L488-L535